workflow/automation/estimation/estimate_cybershake.py

```diff
@@ -189,151 +190,46 @@ def run_estimations(
-
-    print()
-    sum_df = df.sum()
-    header = "{:<12}{:<10}{:<8}".format("core hours", "run time", "")
-    print("{:>14}{:>30}{:>30}".format("LF", "HF", "BB"))
-    print("{:>12}{}{}{}".format("", header, header, header))
-    print(
-        "{:<12}{:<12.3f}{:<10.3f}{:<8.0}"
-        "{:<12.3f}{:<10.3f}{:<8.0}"
-        "{:<12.3f}{:<10.3f}{:<8.0}".format(
-            "Total",
-            sum_df.loc[
-                const.ProcessType.EMOD3D.str_value, const.MetadataField.core_hours.value
-            ],
-            sum_df.loc[
-                const.ProcessType.EMOD3D.str_value, const.MetadataField.run_time.value
-            ],
-            "",
-            sum_df.loc[
-                const.ProcessType.HF.str_value, const.MetadataField.core_hours.value
-            ],
-            sum_df.loc[
-                const.ProcessType.HF.str_value, const.MetadataField.run_time.value
-            ],
-            "",
-            sum_df.loc[
-                const.ProcessType.BB.str_value, const.MetadataField.core_hours.value
-            ],
-            sum_df.loc[
-                const.ProcessType.BB.str_value, const.MetadataField.run_time.value
-            ],
-            "",
+    if im_calc_input_data is not None:
+        print("Running IM_calc estimation")
+        im_calc_core_hours, im_calc_run_time = estimate_wct.est_IM_chours_single(
+            *im_calc_input_data
         )
-    )
-
-
-def get_runs_dir_params(
-    runs_dir: str, fault_names: List[str], realisations: np.ndarray
-):
-    """Gets the parameters from the runs directory. Assumes that all realisations of a
-    fault have the same parameters
-    """
-    data = []
-    for ix, fault_name in enumerate(fault_names):
-        r = realisations[ix][0]
+        im_calc_cores = [im_calc_input_data[-1]] * sum(r_counts)
+    else:
+        im_calc_core_hours, im_calc_run_time, im_calc_cores = np.nan, np.nan, np.nan
 
-        params = utils.load_sim_params(
-            os.path.join(runs_dir, fault_name, r, "sim_params.yaml")
+    results_df[
+        (
+            const.ProcessType.IM_calculation.str_value,
+            const.MetadataField.core_hours.value,
         )
+    ] = im_calc_core_hours
+    results_df[
+        (const.ProcessType.IM_calculation.str_value, const.MetadataField.run_time.value)
+    ] = im_calc_run_time
+    results_df[
+        (const.ProcessType.IM_calculation.str_value, const.MetadataField.n_cores.value)
+    ] = im_calc_cores
 
-        data.append((params.dt, params.hf.dt, params.FD_STATLIST, params.hf.slip))
-
-    return np.rec.array(
-        data,
-        dtype=[
-            ("dt", np.float32),
-            ("hf_dt", np.float32),
-            ("fd_statlist", np.object),
-            ("slip", np.object),
-        ],
-    )
+    return results_df
 
 
-def main(args):
+def main(
+    vms_dir: str,
+    sources_dir: str,
+    runs_dir: str,
+    fault_selection: str = None,
+):
```

Comment:
🥲
Just need to refactor it to take in the root folder, not three separate ones.
Not your job though